README.md

```diff
@@ -138,15 +151,15 @@ example1 list = runState 0 $ do
   put (i + length list)
 ```
 
-`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation. The `get` function is resolved with a visible type application, due to the fact that effects can contain more than one state type (in contrast with `mtl`'s `MonadState`, which limits the user to a single state type).
+`runState` returns a tuple of both the computed value (the `()`) and the final state (the `Int`), visible in the result of the returned computation. The `get` function is resolved with a visible type application, due to the fact that effects can contain more than one state type (in contrast with `mtl`’s `MonadState`, which limits the user to a single state type).
 
 Since this function returns a value in some carrier `m`, effect handlers can be chained to run multiple effects. Here, we get the list to compute the length of from a `Reader` effect:
 
 ```haskell
 example2 :: Algebra sig m => m (Int, ())
 example2 = runReader "hello" . runState 0 $ do
-  list <- ask
-  put (length (list :: String))
+  list <- ask @String
+  put @Int (length list)
```

Comment:
I’m especially skeptical of the type application on `put` since `length` is monomorphic in its result type.

Reply:
I’ve been thinking about it and I think you’re right. I’ll remove the type applications, though I think we should keep the FAQ entry about when they are useful.